models/aether_cube_sphere/transform_state_mod.f90

```diff
@@ -0,0 +1,871 @@
+use utilities_mod,              only : find_namelist_in_file, check_namelist_read,  &
+                                       error_handler, E_ERR, string_to_integer
+
+use cube_sphere_grid_tools_mod, only : lat_lon_to_col_index, get_grid_delta
+
+implicit none
+private
+
+public :: initialize_transform_state_mod,  model_to_dart, dart_to_model, &
+          get_ensemble_range_from_command_line
+
+integer  :: iunit, io
+
+! version controlled file description for error handling, do not edit
+character(len=*), parameter :: source   = 'aether_cube_sphere/transform_state_mod.f90'
+character(len=*), parameter :: revision = ''
+character(len=*), parameter :: revdate  = ''
+
+type :: file_type
+   character(len=256) :: file_path
+   integer            :: ncid, unlimitedDimId, nDimensions, nVariables, nAttributes, formatNum
+end type file_type
+
+! It would be nice to get this information from the Aether input files, not possible for now
+integer            :: np, nblocks, nhalos
+
+! Switch between scalar and horizontal f10.7
+logical  :: scalar_f10_7 = .true.
+
+namelist /transform_state_nml/ np, nblocks, nhalos, scalar_f10_7
+
+contains
+
+!---------------------------------------------------------------
+
+subroutine initialize_transform_state_mod()
+
+! Just read the namelist
+call find_namelist_in_file('input.nml', 'transform_state_nml', iunit)
+read(iunit, nml = transform_state_nml, iostat = io)
+call check_namelist_read(iunit, io, 'transform_state_nml')
+
+end subroutine initialize_transform_state_mod
+
+!---------------------------------------------------------------
+
+subroutine model_to_dart(aether_block_file_dir, dart_file_dir, ensemble_number)
+
+character(len=*), intent(in) :: aether_block_file_dir, dart_file_dir
+integer,          intent(in) :: ensemble_number
+
+integer  :: iblock, dimid, length, ncols, dart_dimid(3), varid, xtype, nDimensions, nAtts
+integer  :: param_dimid(2), nparams
+integer  :: ix, iy, iz, icol, ncstatus
+integer  :: ntimes(nblocks), nxs(nblocks), nys(nblocks)
+integer  :: ions_ntimes(nblocks),     ions_nxs(nblocks),     ions_nys(nblocks)
+integer  :: neutrals_ntimes(nblocks), neutrals_nxs(nblocks), neutrals_nys(nblocks)
+integer  :: haloed_nxs(nblocks), haloed_nys(nblocks)
+integer  :: final_nzs, ions_final_nzs, neutrals_final_nzs
+integer  :: filter_time_id, filter_alt_id, filter_lat_id, filter_lon_id 
+integer  :: grid_alt_id,    grid_lat_id,   grid_lon_id
+integer  :: electron_varid, f10_7_varid
+integer  :: dimids(NF90_MAX_VAR_DIMS)
+real(r8) :: blat, blon, del, half_del, f10_7_val
+real(r8) :: time_array(1)
+logical  :: add_to_electrons
+character(len = 4)           :: ensemble_string
+character(len=NF90_MAX_NAME) :: name, attribute
+integer,         allocatable :: col_index(:, :, :), filter_ions_ids(:), filter_neutrals_ids(:)
+! File for reading in variables from block file; These can be R4
+real(r4),        allocatable :: spatial_array(:), variable_array(:, :, :), electron_array(:, :)
```

Comment:
Are the state restart files always single precision?  Maybe you want to set dart to write single precision output files. 

https://github.com/NCAR/DART/blob/e5cfb8b885344140bf6a5b762611ace0c2fea3d7/assimilation_code/modules/io/state_vector_io_mod.f90#L113-L116